catalog/harvest/catalog-harvest-api/src/main/java/org/codice/ddf/catalog/harvest/Harvester.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest;
+
+/**
+ * A {@code Harvester} is responsible for fetching remote resources and notifying any registered
+ * {@link Listener}s of create, update and delete events.
+ *
+ * <p>In the event that a {@code Harvester} goes down, it should make a best effort to remember the
+ * last state of the external server it was harvesting from. Upon a {@code Harvester} coming back
+ * up, it should notify registered {@link Listener}s of new events.
+ *
+ * <p><b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface Harvester {
+
+  /**
+   * Registers a new {@link Listener} to receive events from the remote source being harvested from.
+   * Newly {@link Listener}s will only receive new events from this {@code Harvester}.
+   *
+   * @param listener the {@link Listener} to register
+   * @throws IllegalArgumentException if the {@link Listener} is already registered or cannot be
```

Comment:
❓ Can registering be idempotent like unregistering is below?

Reply:
I changed this to throw an IllegalArgumentException if the Listener cannot be registered (ie if the Harvester only supports a limited number of Listeners), and to be a no-op if the Listener is already registered.